Kernel/Arch/x86/SIMDState.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2022, Leon Albrecht <leon.a@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/EnumBits.h>
+#include <AK/FPControl.h>
+#include <AK/Platform.h>
+#include <AK/Types.h>
+
+namespace Kernel::SIMD {
+
+// Intel-Manual Vol 1 Chp 13.4
+enum StateComponent : u64 {
+    X87 = 1ull << 0,
+    SSE = 1ull << 1u, // xmm0-xmm7(15)
+    AVX = 1ull << 2,  // ymm0-ymm7(15) hi
+    MPX_BNDREGS = 1ull << 3,
+    MPX_BNDCSR = 1ull << 4,
+    AVX512_opmask = 1ull << 5, // k0 - k9
+    AVX512_ZMM_hi = 1ull << 6, // 0 - 15
+    AVX512_ZMM = 1ull << 7,    // 16 - 31 full
+    PT = 1ull << 8,
+    PKRU = 1ull << 9,
+
+    CET_U = 1ull << 11,
+    CET_S = 1ull << 12,
+    HDC = 1ull << 13,
+
+    LBR = 1ull << 15,
+    HWP = 1ull << 16,
+
+    XCOMP_ENABLE = 1ull << 63
+};
+AK_ENUM_BITWISE_OPERATORS(StateComponent);
+
+struct [[gnu::packed]] LegacyRegion {
+    AK::X87ControlWord FCW;
+    u16 FSW;
+    u8 FTW;
+    u8 : 8;
+    u16 FOP;
+#if ARCH(I386)
+    struct {
+        // 32-bit version
+        u32 FIP_32;
+        u16 FCS;
+        u16 : 16;
+        u32 FPD_32;
+        u16 FDS;
+        u16 : 16;
+    };
+#elif ARCH(X86_64)
+    struct {
+        // 64-bit version
+        u64 FIP_64;
+        u64 FDP_64;
+    };
```

Comment:
Hmm, what's the point of the naked structs?

Reply:
You'll have to ask @Hendiadyoin1 about this one. 
I believe this is to change the way the members are accessed depending on the architecture, and to actually enforce it. These members _are_ a part of the legacy area, but just interpreted differently between ia32 and amd64 targets.

Reply:
There are in a struct, because in the previous revision they were part of a union,
They can now safely be put into the upper struct